tests/e2e/vsphere_volume_expansion.go

```diff
@@ -412,6 +396,473 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+		8. Bring up SPSservice
+		9. Verify the resized PVC by doing CNS query
+		10. Make sure data is intact on the PV mounted on the pod
+		11. Make sure file system has increased
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify online volume expansion when SPS-Service is down ", func() {
+
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		volHandle, pvclaim, pv, storageclass := createSCwithVolumeExpansionTrueAndDynamicPVC(f, client, "", namespace)
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pv.Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		pod := createPODandVerifyVolumeMount(f, client, namespace, pvclaim, volHandle)
+		originalFSSize, err := getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Verify volume is detached from the node")
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		}()
+
+		ginkgo.By("Bring down SPS service")
+		isSPSServiceStopped := true
+		vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+		err = invokeVCenterServiceControl(stopOperation, spsServiceName, vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			if isSPSServiceStopped {
+				framework.Logf("Bringing sps up before terminating the test")
+				err = invokeVCenterServiceControl(startOperation, spsServiceName, vcAddress)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Expanding current pvc")
+		currentPvcSize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		newSize := currentPvcSize.DeepCopy()
+		newSize.Add(resource.MustParse("1Gi"))
+		framework.Logf("currentPvcSize %v, newSize %v", currentPvcSize, newSize)
+		pvclaim, err = expandPVCSize(pvclaim, newSize, client)
+		framework.ExpectNoError(err, "While updating pvc for more size")
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		ginkgo.By("File system resize should not succeed Since SPS service is down. Expect an error")
+		expectedErrMsg := "ServerFaultCode: CnsFault"
```

Comment:
This will be present for any cnsFault right? I would suggest to use the correct error string.

Reply:
there is no specific error code like we get when we bring down VSAN health. 
For SPS , This is the error message we get " error: ServerFaultCode: CnsFault"

Reply:
what is the full error, may be we can use a different part of the message

Reply:
The Full error message is like this " resize volume "pvc-1735941b-30f5-417d-9a2d-9be7c433bf7a" by resizer "csi.vsphere.vmware.com" failed: rpc error: code = Internal desc = failed to expand volume: "b3c68250-1dc2-493c-8b77-34fdc0fdc050" to size: 2048 with error: ServerFaultCode: CnsFault"

Reply:
First part of the message is also common "pvc-1735941b-30f5-417d-9a2d-9be7c433bf7a" by resizer "csi.vsphere.vmware.com" failed: rpc error: code = Internal desc = failed to expand volume: "b3c68250-1dc2-493c-8b77-34fdc0fdc050" 
only message differs is "ServerFaultCode: CnsFault" hence used the same